src/main/java/org/takes/tk/TkCORS.java

```diff
@@ -0,0 +1,119 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.takes.tk;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import lombok.EqualsAndHashCode;
+import org.takes.Request;
+import org.takes.Response;
+import org.takes.Take;
+import org.takes.rs.RsWithHeaders;
+import org.takes.rs.RsWithStatus;
+
+/**
+ * CORS take.
+ *
+ * <p>This take checks if the request (Origin) is allowed to perform
+ * the desired action against the list of the given domains.</p>
+ *
+ * <p>The specification of CORS can be found on the W3C web site on the
+ * following <a href="http://www.w3.org/TR/cors/">link</a> or even on the <a
+ * href="https://tools.ietf.org/html/rfc6454">RFC-6454</a> specification.
+ *
+ * @author Endrigo Antonini (teamed@endrigo.com.br)
+ * @version $Id$
+ * @since 0.20
+ */
+@EqualsAndHashCode(of = { "origin" , "allowed" })
+public final class TkCORS implements Take {
+
+    /**
+     * Prefix of HTTP head Origin.
+     */
+    private static final String ORIGIN_PREFIX = "Origin:";
+
+    /**
+     * Original take.
+     */
+    private final transient Take origin;
+
+    /**
+     * List of allowed domains.
+     */
+    private final transient Set<String> allowed;
+
+    /**
+     * Ctor.
+     * @param take Original
+     * @param domains Allow domains
+     */
+    public TkCORS(final Take take, final String... domains) {
+        this.origin = take;
+        this.allowed = new HashSet<String>();
+        if (domains != null) {
+            this.allowed.addAll(Arrays.asList(domains));
+        }
+    }
+
+    @Override
+    public Response act(final Request req) throws IOException {
+        boolean hasorigin = false;
+        Response response;
+        String domain = "";
+        for (final String head : req.head()) {
+            if (head.startsWith(ORIGIN_PREFIX)) {
+                hasorigin = true;
+                domain = head.split(ORIGIN_PREFIX)[1].trim();
+                break;
+            }
+        }
+        if (hasorigin && (
+            this.allowed.isEmpty()
+            || this.allowed.contains(domain))) {
+            response = this.origin.act(req);
+            final Set<String> headers = new HashSet<String>();
+            headers.add("Access-Control-Allow-Credentials: true");
+            // @checkstyle LineLengthCheck (1 line)
+            headers.add("Access-Control-Allow-Methods: OPTIONS, GET, PUT, POST, DELETE, HEAD");
```

Comment:
@yegor256 How you suggest that? Just breaking the parameter to the next link doesn't solve the problem, it continues with more than 80 chars. Do you suggest using `String.format` and breaking it in small strings?